### PR TITLE
Feature/course employment stats

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

Refactored to move common stats functionality into shared class. Implemented Employment stats.

### How to review

Review changes/

### CODE REVIEW

File: EtlPipelineBlobTrigger/course_stats.py [hunk 1]
```diff
@@ -129,10 +187,58 @@ def get_raw_continuation_list(raw_course_data):
...
+        return self.shared_utils.get_unavailable(xml_elem, subj_key, agg_key,
+                                                 unavail_reason_key)
+
+    def get_employment(self, raw_course_data):
+        subj_key = 'EMPSBJ'
+        agg_key = 'EMPAGG'
+        unavail_reason_key = 'EMPUNAVAILREASON'
+        element_key = 'EMPLOYMENT'
+
+        json_elem_list = []
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/course_stats.py]: This structure can be reused for get_continuation?

File: EtlPipelineBlobTrigger/course_stats.py [hunk 2]
```diff
@@ -1,105 +1,163 @@
...
+    def get_stats(self, raw_course_data):
+        stats = {}
+
+        continuation = Continuation()
+        employment = Employment()
+        stats['continuation'] = continuation.get_continuation(raw_course_data)
+        stats['employment'] = employment.get_employment(raw_course_data)
+        return stats
+
+
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/course_stats.py]: So I generally find it easier to follow code if the classes and the encapsulated functions are in logical flow from high level to lower levels of abstracted code.

```
class CourseStats:
  ...
class Continuation:
  ...
class Employment:
  ...
class SharedUtils:
  ...
```

File: EtlPipelineBlobTrigger/course_stats.py (hunk 1 above)
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: Yes, as discussed planning on more refactoring on next story.

File: EtlPipelineBlobTrigger/course_stats.py (hunk 2 above)
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: :+1 Moved SharedUtils to bottom of module.

File: EtlPipelineBlobTrigger/course_stats.py (hunk 1 above)
Comment (nshumoogum) [reply EtlPipelineBlobTrigger/course_stats.py]: Good good

**Pull request merged**